pubsub.go

```diff
@@ -64,18 +64,13 @@ func (ps *PubSub) Publish(m Message) error {
 	topic := ps.client.Topic(m.Topic)
 	topic.PublishSettings = ps.settings.publish.Settings
 
-	data, err := json.Marshal(m.Message)
-	if err != nil {
-		return err
-	}
-
 	ctx := context.Background()
 	result := topic.Publish(ctx, &pubsub.Message{
-		Data:       data,
+		Data:       []byte(m.Message.(string)),
```

Comment:
When we don't marshal the json object and just do a `toString` conversion, the string object won't have the defined fields which will make unmarshaling it difficult for the consumer. What happens right now when you marshal a string that contains the xml contents?